app/pages/patientdata/patientdata.js

```diff
@@ -776,12 +776,13 @@ export let PatientData = translate()(React.createClass({
     const pdfGenerating = nextProps.generatingPDF;
     const pdfGenerated = _.get(nextProps, 'pdf.combined', false);
     const patientDataProcessed = (!nextState.processingData && !!nextState.processedPatientData);
+    const userFetched = !nextProps.fetchingUser;
     const hasDiabetesData = _.get(nextState, 'processedPatientData.diabetesData.length');
 
     // Ahead-Of-Time pdf generation for non-blocked print popup.
     // Whenever patientData is processed or the chartType changes, such as after a refresh
     // we check to see if we need to generate a new pdf to avoid stale data
-    if (patientDataProcessed && hasDiabetesData && !pdfGenerating && !pdfGenerated) {
+    if (userFetched && patientDataProcessed && hasDiabetesData && !pdfGenerating && !pdfGenerated) {
```

Comment:
I'm curious.  Were you running into any issues not having this check here?  I can't recall anything from the user object being leveraged in the PDFs.

Reply:
It's to load preferences and the language. When it was stored in `profile.language` there were no issues as it's loaded with the patient, but the preferences are loaded later and as such `preferences.displayLanguageCode`.

Without `userFetched`, the generated pdf doesn't take into account the language, unless for example the user goes to theirr profile and back to the data.